test/stablecoin_converter.js

```diff
@@ -0,0 +1,110 @@
+const StablecoinConverter = artifacts.require("StablecoinConverter")
+const StablecoinConverterTestInterface = artifacts.require("StablecoinConverterTestInterface")
+
+const IdToAddressBiMap = artifacts.require("IdToAddressBiMap")
+const ERC20 = artifacts.require("ERC20")
+
+const truffleAssert = require("truffle-assertions")
+const {
+  waitForNSeconds,
+  sendTxAndGetReturnValue } = require("./utilities.js")
+
+
+contract("StablecoinConverter", async (accounts) => {
+
+  const [user_1] = accounts 
+  beforeEach(async () => {
+    const lib1 = await IdToAddressBiMap.new()
+        
+    await StablecoinConverter.link(IdToAddressBiMap, lib1.address)
+    await StablecoinConverterTestInterface.link(IdToAddressBiMap, lib1.address)
+
+  })
+
+  describe("placeOrder", () => { 
+    it("places Orders and checks parameters", async () => {
+      const stablecoinConverter = await StablecoinConverter.new()
+      const currentStateIndex = await stablecoinConverter.getCurrentStateIndex()
+      const id = await stablecoinConverter.placeOrder.call(0, 1, true, 3, 10, 20, {from: user_1})
+      await stablecoinConverter.placeOrder(0, 1, true, 3, 10, 20, {from: user_1})
+      assert.equal(((await stablecoinConverter.orders.call(user_1,id)).sellAmount).toNumber(), 20, "sellAmount was stored incorrectly")
```

Comment:
nit, maybe just `await stablecoinConverter.orders.call(user_1,id))` once and then assert on the fetched result, rather than calling 7 times.